### PR TITLE
Add pybip39

### PR DESCRIPTION
**Pull request by MementoRC**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MementoRC** [comment]
Could adding python in the build section resolve the issue? It looks like python is found in order:
   1 PYO3_PYTHON
   2 venv or conda PREFIX
   3 search in bin directories

There is also an [env] in Cargo.toml, but it is likely not dynamic - I don't have a windows system, so I may go debug-commit mode :\

File: recipes/pybip39/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
+    - {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
+    - cargo-bundle-licenses
+    - python
```
Comment (xhochy) [inline recipes/pybip39/meta.yaml:27]: This should not be here and will likely lead to break, rather try setting `PYO3_PYTHON` for windows.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/pybip39/meta.yaml (hunk 1 above)
Comment (MementoRC) [reply recipes/pybip39/meta.yaml]: It passes though :'(

Do you think that this should work:
```
  script_env:
    - PYO3_PYTHON: %PREFIX%\bin\python.exe  # [win]
```
or:
```
  script_env:
    - PYO3_PYTHON: {{ PYTHON }}  # [win]
```
Well, the latter should work

Comment (xhochy) [reply recipes/pybip39/meta.yaml]: Yes, try that. The other one might lead to an empty package at one point.

**Review by xhochy — APPROVED** [review]

**Pull request merged**